### DISCUSSION TITLE
Record Locking Mint Vanessa

### DISCUSSION THREAD

**Discussion by knightflyer** [ltsp/ltsp#781] [General]
NFS record locking does not work between ltsp clients with my current configuration.
However works perfectly between ltsp client and server or other nfs connected hosts.
Getting ready to implement my client's big upgrade after a year. Glad I found this before upgrading the client.

NAME="Linux Mint"
VERSION="21 (Vanessa)"
ID=linuxmint
ID_LIKE="ubuntu debian"
PRETTY_NAME="Linux Mint 21"
VERSION_ID="21"
UBUNTU_CODENAME=jammy

Here are my server's /etc/exports entries:

/u    192.168.1.0/24(rw,sync,all_squash,anonuid=1001,anongid=1001,no_subtree_check)

My /etc/ltsp/ltsp.conf entry :

FSTAB_x="192.168.1.200:/u /u nfs defaults 0 0"

Thank you everyone for the help.

**Comment by alkisg** [comment]
Hi @knightflyer, in this case, LTSP just puts the FSTAB line in /etc/fstab; its role ends there.
So whatever issues exist after that, are not related to LTSP, you may troubleshoot them separately, by asking for help in LinuxMint forums / IRC channels etc.

That said, make sure that you have the `nfs-common` package in the LTSP image.

**Comment by knightflyer** [comment]
Greetings thank you for the reply

I respectfully disagree.  I have stand alone hosts runnning the same LINUX
as the LTSP/NFS server that also mount that LTSP server NFS share using the
same entry in the /etc/ltsp/ltsp.conf.
Locking works correctly between the standalone hosts and the LTSP clients,
however when 2 connected LTSP clients access the same record, no lock
occurs and no update occurs.

Let me know if I can do anything to help.

**Comment by alkisg** [comment]
Hi knightflyer,

when replying via email to github, please do not include the previous converstation, to avoid cluttering  https://github.com/ltsp/ltsp/discussions/781.

1) Run `mount | grep nfs` in an ltsp clients and in a standalone client.
Paste the line for /u here, so that we can compare if it's the same or not.

2) Can you provide a test case? E.g. after the clients have booted and the mounts are mounted, can you provide a series of commands to verify if NFS record locking works or not?

**Comment by knightflyer** [comment]
## The following is mount output from LTSP client 1

192.168.1.200:/u on /u type nfs4
(rw,relatime,vers=4.2,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=192.168.1.6,local_lock=none,addr=192.168.1.200)

## The following is mount output from the stand alone client

192.168.1.200:/u on /u type nfs4
(rw,relatime,vers=4.2,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=192.168.1.21,local_lock=none,addr=192.168.1.200)


## When these 2 machines try and access the same record the lock mechanism
works

## The following is mount output from LTSP client 2
192.168.1.200:/u on /u type nfs4
(rw,relatime,vers=4.2,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=192.168.1.5,local_lock=none,addr=192.168.1.200)


## When the 2 LTSP clients try and access the same record the locking
mechanism fails allowing multiple users to collide and allow corrupt data.

The way I test locking
on first client I load an inventory record
If the second client trys to access the same record it should detect the
lock and display error message "Waiting"
once the first client releases the record the second client gets access.

Everything works except if they are both LTSP clients.

**Comment by alkisg** [comment]
> on first client I load an inventory record

1) Which program are you talking about?
2) Where is the lock created, in /u which is NFS, or in /home/username which is SSHFS?
3) Does the program work if you use NFS home instead of SSHFS home?

**Comment by knightflyer** [comment]
Sculptor is 4GL application development package based on isam records.
The lock is created in /u  all the shared data and documents are in /u
I'm not sure how to use NFS home
I changed /etc/ltsp/ltsp.conf
#FSTAB_HOME="server:/home /home nfs defaults,nolock 0 0"
FSTAB_HOME="192.168.1.200:/home /home nfs defaults,nolock 0 0"

and changed the /etc/exports to:
/u
192.168.1.0/24(rw,sync,all_squash,anonuid=1001,anongid=1001,no_subtree_check)
/home
192.168.1.0/24(rw,sync,all_squash,anonuid=1001,anongid=1001,no_subtree_check)

Only 1 user works now trying to logon the LTSP client
So I tried adding a new user and reimaging.
All users just flip back to logon screen except one.
I logged onto both LTSP clients using that user
Locking still doesn't work.
Going to set back and see if my users start working again.

**Comment by knightflyer** [comment]
Okay I studied on how to correctly export /home using nfs
I logged in as before on 2 ltsp clients running /home connected with nfs.
I checked the mount and it shows it is mounting /home/nfs as /home

Still locking fails.  the ltsp clients don't share the lock between each
other, but will share the lock with other non ltsp clients.
The lock mechanism works correctly if trying to access the record multiple
times on the same ltsp client.
